src/coreclr/src/jit/layout.cpp

```diff
@@ -453,7 +451,10 @@ bool ClassLayout::AreCompatible(const ClassLayout* layout1, const ClassLayout* l
         return true;
     }
 
+    assert(clsHnd1 != NO_CLASS_HANDLE);
```

Comment:
Why did the asserts move down? Seems strange to allow clsHnd2 == NO_CLASS_HANDLE for the above checks

Reply:
We could have a situation when we have `GT_BLOCK <40>(GT_ADDR ref (LCL_VAR struct<40>))`. `GenTreeBlock` nodes do not have a class handle, they represent a memory chunk. Such chunk is compatible with a struct if they have the same size and the struct does not have GC pointers.